ConfigurationError raises an Exception

**Describe the bug**
Configuration error, when raised, instead raises a `TypeError`.

**To Reproduce**
Steps to reproduce the behavior:
1. Add a new test in `scripts/tests/twister/test_<any>.py`, which raises a ConfigurationError
2. Run the said test with `pytest`
3. Notice the error name.

**Expected behavior**
Raising `ConfigurationError` bubbles up as a `ConfigurationError` the whole way through.

**Impact**
`ConfigurationError` is unusable, except for naively checking whether *any* Exception has been thrown.

**Logs and console output**
```
    def __init__(self, cfile, message):
>       TwisterException.__init__(self, cfile + ": " + message)
E       TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'

pylib/twister/twisterlib/error.py:16: TypeError
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: N/A
 - Commit SHA: `92fb90c483e09836d189c7cb2bb6464f53c6489d`

--- a/scripts/pylib/twister/twisterlib/error.py
+++ b/scripts/pylib/twister/twisterlib/error.py
@@ -13,7 +13,7 @@ class TwisterRuntimeError(TwisterException):
 
 class ConfigurationError(TwisterException):
     def __init__(self, cfile, message):
-        TwisterException.__init__(self, cfile + ": " + message)
+        TwisterException.__init__(self, str(cfile) + ": " + message)
 
 
 class BuildError(TwisterException):

--- a/scripts/tests/twister/test_twister.py
+++ b/scripts/tests/twister/test_twister.py
@@ -16,6 +16,7 @@ ZEPHYR_BASE = os.getenv("ZEPHYR_BASE")
 sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/pylib/twister"))
 
 import scl
+from twisterlib.error import ConfigurationError
 from twisterlib.testplan import TwisterConfigParser
 
 def test_yamlload():
@@ -77,3 +78,16 @@ def test_testsuite_config_files():
 
     # Check extra kconfig statements, too
     assert scenario["extra_configs"] == ["CONFIG_FOO=y"]
+
+def test_configuration_error():
+    """Test to validate that the ConfigurationError is raisable without further errors.
+
+    It ought to have an str with a path, colon and a message as its only args entry.
+    """
+    filename = Path(ZEPHYR_BASE) / "scripts/tests/twister/test_twister.py"
+
+    with pytest.raises(ConfigurationError) as exception:
+        raise ConfigurationError(filename, "Dummy message.")
+
+    assert len(exception.value.args) == 1
+    assert exception.value.args[0] == str(filename) + ": " + "Dummy message."

